### PR TITLE
docs: expand on how jest version is determined and common workarounds

### PR DESCRIPTION
**Pull request by G-Rath**
This is something of a documentation stop-gap while I work on #889.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,22 +60,40 @@ This is included in all configs shared by this plugin, so can be omitted if
 extending them.
 
 The behaviour of some rules (specifically `no-deprecated-functions`) change
```
Comment (SimenB) [inline README.md:62]: we should create a section here and link to it from https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/no-deprecated-functions.md

**Review by SimenB — CHANGES_REQUESTED** [review]

**Comment by G-Rath** [comment]
Currently this is actually incorrect, since we assume that the version being passed in `settings` is the jest major version, which `require('jest/package.json').version` won't be - so we should either land the `parseJestVersion` function from #889 as a fix or otherwise merge this after #889 (which'd mean this should be adjusted to explain the new logic).

File: docs/rules/no-deprecated-functions.md [hunk 2]
```diff
@@ -6,6 +6,10 @@ either been renamed for clarity, or replaced with more powerful APIs.
 While typically these deprecated functions are kept in the codebase for a number
 of majors, eventually they are removed completely.
 
+This rule requires knowing which version of Jest you're using - see
+[this section of the readme](README.md) for details on how that is obtained
```
Comment (SimenB) [inline docs/rules/no-deprecated-functions.md:10]: This link doesn't work, does it?

Comment (G-Rath) [reply docs/rules/no-deprecated-functions.md:10]: No it doesn't - IntelliJ seems to have weird handling of relative links in nested docs :/

**Comment by G-Rath** [comment]
Have pulled these changes into #889

**Pull request closed without merging**